scripts/lib/custom_networking.sh

```diff
@@ -0,0 +1,65 @@
+func configure_custom_networking(){
+    local VPC_ID=$(aws eks describe-cluster --name "$CLUSTER_NAME" | grep -iwo 'vpc-[a-zA-z0-9]*' | xargs)
+    echo "VPC $VPC_ID"
+   
+    #associate an additional CIDR block with the range 100.64.0.0/16 to the VPC
+    aws ec2 associate-vpc-cidr-block --vpc-id $VPC_ID --cidr-block 100.64.0.0/16
+
+    local AZ1=us-west-2a
+    local AZ2=us-west-2b
+    local AZ3=us-west-2c
+
+    CUST_SNET1=$(aws ec2 create-subnet --cidr-block 100.64.0.0/19 --vpc-id $VPC_ID --availability-zone $AZ1 | jq -r .Subnet.SubnetId)
+    CUST_SNET2=$(aws ec2 create-subnet --cidr-block 100.64.32.0/19 --vpc-id $VPC_ID --availability-zone $AZ2 | jq -r .Subnet.SubnetId)
+    CUST_SNET3=$(aws ec2 create-subnet --cidr-block 100.64.64.0/19 --vpc-id $VPC_ID --availability-zone $AZ3 | jq -r .Subnet.SubnetId)
+
+    echo "Subnets - $CUST_SNET1 $CUST_SNET2 $CUST_SNET3"
+    aws ec2 create-tags --resources $CUST_SNET1 --tags Key=kubernetes.io/cluster/$CLUSTER_NAME,Value=shared
+    aws ec2 create-tags --resources $CUST_SNET2 --tags Key=kubernetes.io/cluster/$CLUSTER_NAME,Value=shared
+    aws ec2 create-tags --resources $CUST_SNET3 --tags Key=kubernetes.io/cluster/$CLUSTER_NAME,Value=shared
+
+    local RTASSOC_ID=$(aws ec2 describe-route-tables --filters Name=vpc-id,Values=$VPC_ID |jq -r '.RouteTables[].RouteTableId')
+
+    echo "RT table $RTASSOC_ID"
+
+    aws ec2 associate-route-table --route-table-id $RTASSOC_ID --subnet-id $CUST_SNET1
+    aws ec2 associate-route-table --route-table-id $RTASSOC_ID --subnet-id $CUST_SNET2
+    aws ec2 associate-route-table --route-table-id $RTASSOC_ID --subnet-id $CUST_SNET3
+
+    #Enable custom networking
+    $KUBECTL_PATH set env daemonset aws-node -n kube-system AWS_VPC_K8S_CNI_CUSTOM_NETWORK_CFG=true
+    sleep 5
```

Comment:
Sleep is not very robust. This will result in flaky tests. Are you trying to check if the daemonset is ready?

You could have helper functions like below and use those to check the rollout completion

```
# resource_exists returns 0 when the supplied resource can be found, 1
# otherwise. An optional second parameter overrides the Kubernetes namespace
# argument
k8s_resource_exists() {
    local __res_name=${1:-}
    local __namespace=${2:-}
    local __args=""
    if [ -n "$__namespace" ]; then
        __args="$__args-n $__namespace"
    fi
    kubectl get $__args "$__res_name" >/dev/null 2>&1
}


# check_deployment_rollout watches the status of the latest rollout
# until it's done or until the timeout. Namespace and timeout are optional
# parameters
check_deployment_rollout() {
    local __dep_name=${1:-}
    local __namespace=${2:-}
    local __timeout=${3:-"2m"}
    local __args=""
    if [ -n "$__namespace" ]; then
        __args="$__args-n $__namespace"
    fi
    kubectl rollout status deployment/"$__dep_name" $__args --timeout=$__timeout
}

```

Reply:
Yeah I just added a sleep to give a little bit delay but later in the code I have this to wait to confirm all pods are up - 
```
while [[ $($KUBECTL_PATH describe ds aws-node -n=kube-system | grep "Available Pods: 0") ]]
    do
        sleep 5
        echo "Waiting for daemonset update"
    done
    echo "Updated!
```

I can move this logic into the helper function which you shared and we can reuse it in integration script and other newer test scripts.